### PR TITLE
Migrate to oras-go library

### PR DESCRIPTION
**Pull request by ndeloof**
see https://github.com/deislabs/oras/pull/265

**What this PR does / why we need it**:
https://github.com/deislabs/oras has been rearchitectured to introduce oras-go library
This PR adopts this library

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by jdolitsky — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -34,7 +34,7 @@ require (
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.7.0
 	github.com/xeipuuv/gojsonschema v1.2.0
-	golang.org/x/crypto v0.0.0-20210220033148-5ea612d1eb83
+	golang.org/x/crypto v0.0.0-20210322153248-0c34fe9e7dc2
```
Comment (bacongobbler) [inline go.mod:37]: Is there a way to prevent this update in this PR? I'm trying to figure out what's necessary for the name change vs. what should be pulled out into a separate PR. Changes to /x/crypto might be best as a separate review item if at all possible.

Comment (ndeloof) [reply go.mod:37]: I can't tell, this has been introduced by `go mod tidy`, so I assume required by transitive dependencies...

Comment (hickeyma) [reply go.mod:37]: @ndeloof I agree with @bacongobbler. Lets not change modules that are not part of the PR in question. Do you mind rolling this back?

Comment (ndeloof) [reply go.mod:37]: As I said, this update is forced by `go mod`, if I revert this it will be added again any time you run `go mod xx` commands.

Comment (hickeyma) [reply go.mod:37]: @ndeloof Looking further https://github.com/oras-project/oras-go/blob/main/go.mod is added in https://github.com/oras-project/oras-go/blob/main/go.mod, so probably picking it up from there.

**Review by hickeyma — APPROVED** [review]
I am ok with the change to `golang.org/x/crypto` module as stated in https://github.com/helm/helm/pull/9675#discussion_r643802199. LGTM. I will hold on merging as would like @bacongobbler to give his opinion on it.

**Comment by thaJeztah** [comment]
ping @bacongobbler, ptal 🙏

**Comment by ndeloof** [comment]
just rebased to fix go.sum conflict
@bacongobbler?

**Review by bacongobbler — APPROVED** [review]

**Comment by howardjohn** [comment]
A heads up - this change causes an indirect import of k8s.io/kubernetes library which is generally pretty heavy. Opened https://github.com/containerd/containerd/issues/5781 to track this
